vault/auth.go

```diff
@@ -457,6 +461,9 @@ func (c *Core) teardownCredentials() error {
 // newCredentialBackend is used to create and configure a new credential backend by name
 func (c *Core) newCredentialBackend(
 	t string, sysView logical.SystemView, view logical.Storage, conf map[string]string) (logical.Backend, error) {
+	if alias, ok := credentialAliases[t]; ok {
```

Comment:
@jefferai Yes, the comment was for a different thread. It was to decide the default value for `auth_type` field on the `role/roleName` endpoint. Along with mounting, if the scripts are also creating roles (which is likely), now the created roles will default to `iamAuthType`, which both joel and I recall that it was discussed about but there wasn't a conclusion. Joel has actually changed the default to `iamAuthType` and I was wondering if this is okay.